master/cmd/determined-master/root.go

```diff
@@ -193,18 +193,21 @@ func applyBackwardsCompatibility(configMap map[string]interface{}) (map[string]i
 	vProvisioner, provisionerExisted := configMap["provisioner"]
 
 	// Ensure we use either the old schema or the new one.
-	if (rmExisted || rpsExisted) && (schedulerExisted || provisionerExisted) {
+	oldRMConfig := schedulerExisted || provisionerExisted
+	newRMConfig := rmExisted || rpsExisted
+	if newRMConfig && oldRMConfig {
 		return nil, errors.New(
 			"cannot use the old and the new configuration schema at the same time",
 		)
 	}
-	if rmExisted || rpsExisted {
+	if !oldRMConfig {
+		// Use configMap if RMs are not defined at all, or if they are defined using the new schema.
```

Comment:
I have mixed feelings on this change. I think it's the right thing to do, but I'm suspicious of unintended consequences.

Reply:
i agree on both fronts. i would just run a cluster before and after this change, call `det master config` for both, and diff the results. then maybe post it here if there is anything concerning.

Reply:
Thanks for the suggestion! There's only one line different between the configs with/without this specific change. The "new" / proposed version has the addition of this line:
```max_cpu_containers_per_agent: -1```

Which looks like it's probably fine to me.

Reply:
Yeah, the -1 sentinel value seems right to me.